gammapy/makers/utils.py

```diff
@@ -422,3 +424,161 @@ def make_theta_squared_table(
     table.meta["ON_RA"] = position.icrs.ra
     table.meta["ON_DEC"] = position.icrs.dec
     return table
+
+
+def get_rad_max_vs_energy(geom, observation):
+    """Obtain the values of `RAD_MAX` values corresponding to the observation
+    offset and to the bin centers of the energy axis in the `RegionGeom`.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.RegionGeom`
+        Reference map geom.
+    observation : `~gammapy.data.Observation`
+        Observation container.
+
+    Returns
+    -------
+    array : `~astropy.units.Quantity`
+        Values of the `RAD_MAX` corresponding to each estimated energy bin center.
+    """
+    if observation.rad_max is None:
+        raise ValueError(
+            "the IRF for this observation does not include a RAD_MAX_2D table"
+        )
+
+    on_center = geom.center_skydir
+    offset = on_center.separation(observation.pointing_radec)
+
+    rad_max = observation.rad_max.evaluate(
+        offset=offset, energy=geom.axes["energy"].center
+    )
+
+    return rad_max
+
+
+def make_counts_rad_max(geom, observation):
+    """Extract the counts using for the ON region size the values in the
+    `RAD_MAX_2D` table.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.RegionGeom`
+        Reference map geom.
+    observation : `~gammapy.data.Observation`
+        Observation container.
+
+    Returns
+    -------
+    counts : `~gammapy.maps.RegionNDMap`
+        Counts vs estimated energy extracted from the ON region.
+    """
+    rad_max = get_rad_max_vs_energy(geom, observation)
+
+    counts_list = []
+    events = observation.events
+
+    # create and fill a map per each energy bin, fetch the counts
+    for i, rad in enumerate(rad_max):
+        on_region = CircleSkyRegion(center=geom.center_skydir, radius=rad)
+        energy_range = geom.axes["energy"].slice(i)
+        on_region_geom = RegionGeom(on_region, axes=[energy_range])
+        counts = Map.from_geom(on_region_geom)
+        counts.fill_events(events)
+        counts_list.append(counts.data[0])
+
+    counts = RegionNDMap.from_geom(geom, data=np.asarray(counts_list))
+
+    return counts
+
+
+def make_counts_off_rad_max(
```

Comment:
Same here maybe remove the `observation` container from the signature and rely on the individual classes instead..

Reply:
Changed the function's signature from `make_counts_off_rad_max(geom, observation)` to `make_counts_off_rad_max(geom, rad_max, events)`